lntest/itest/lnd_payment_test.go

```diff
@@ -581,24 +581,28 @@ func testBidirectionalAsyncPayments(net *lntest.NetworkHarness, t *harnessTest)
 
 	// Next query for Bob's and Alice's channel states, in order to confirm
 	// that all payment have been successful transmitted.
-	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
-	bobInfo, err := getChanInfo(ctxt, net.Bob)
-	if err != nil {
-		t.Fatalf("unable to get bob's channel info: %v", err)
-	}
+	err = wait.NoError(func() error {
+		ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+		bobInfo, err := getChanInfo(ctxt, net.Bob)
+		if err != nil {
+			t.Fatalf("unable to get bob's channel info: %v", err)
+		}
 
-	if bobInfo.LocalBalance != bobAmt {
-		t.Fatalf("bob's local balance is incorrect, got %v, expected"+
-			" %v", bobInfo.LocalBalance, bobAmt)
-	}
-	if bobInfo.RemoteBalance != aliceAmt {
-		t.Fatalf("bob's remote balance is incorrect, got %v, "+
-			"expected %v", bobInfo.RemoteBalance, aliceAmt)
-	}
-	if len(bobInfo.PendingHtlcs) != 0 {
-		t.Fatalf("bob's pending htlcs is incorrect, got %v, "+
-			"expected %v", len(bobInfo.PendingHtlcs), 0)
-	}
+		if bobInfo.LocalBalance != bobAmt {
+			t.Fatalf("bob's local balance is incorrect, got %v, "+
```

Comment:
if we `t.Fatalf` in every check, won't think only run once anyway? Would thing we need to return these errors rather

Reply:
Thanks, somehow forgot to update these to `Errorf`... Now fixed :sweat_smile: